auth/authenticators/basic-authenticator.ts

```diff
@@ -17,44 +17,63 @@
 import extend = require('extend');
 import { computeBasicAuthHeader, validateInput } from '../utils';
 import { Authenticator } from './authenticator';
-import { AuthenticateOptions, AuthenticatorInterface } from './authenticator-interface';
+import { AuthenticateOptions } from './authenticator-interface';
 
+/** Configuration options for basic authentication. */
 export type Options = {
+  /** The username to be used in basic authorization. */
   username: string;
+  /** The password to be used in basic authorization. */
   password: string;
 }
 
-export class BasicAuthenticator extends Authenticator implements AuthenticatorInterface {
+
+/**
+ * The BasicAuthenticator is used to add basic authentication information to
+ *   requests.
+ *
+ * Basic Authorization will be sent as an Authorization header in the form:
+ *
+ *     Authorization: Basic <encoded username and password>
+ *
+ */
+export class BasicAuthenticator extends Authenticator {
   protected requiredOptions = ['username', 'password'];
-  private username: string;
-  private password: string;
+
+  protected authHeader: {
+    Authorization: string;
+  };
 
   /**
-   * Basic Authenticator Class
-   *
-   * Handles the Basic Authentication pattern.
+   * Create a new BasicAuthenticator instance.
    *
-   * @param {Object} options
-   * @param {String} options.username
-   * @param {String} options.password
-   * @constructor
+   * @param {object} options Configuration options for basic authentication.
+   * @param {string} options.username The username portion of basic authentication.
+   * @param {string} options.password The password portion of basic authentication.
+   * @throws {Error} The configuration options are not valid.
```

Comment:
I vaguely recall a discussion about using `object` vs `Object` but don't remember the details . We should make sure we are consistent on that here, and for `string` vs `String`.

Reply:
`object` vs `Object` matters for TypeScript in the code. It may not matter for documentation as much but it order to not imply the wrong TypeScript meaning, we should probably stick with lowercase: `object` and `string`.